connection/src/main/java/com/netcracker/Connection.java

```diff
@@ -0,0 +1,49 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+public class Connection implements Closeable {
+    private final Socket socket;
+    private final ObjectOutputStream outputStream;
+    private final ObjectInputStream inputStream;
+
+    public Connection(Socket socket) throws IOException {
```

Comment:
&#x1F34E; **Issue**
Is IOException appropriate here ?
Some of your methods rethrow it, others wrap it in RuntimeEception. I think you should choose one of these approaches and not combine them.